### PR TITLE
Destroy product promotion rules on destruct

### PR DESCRIPTION
**Pull request by tommiller1**

### CODE REVIEW

**Comment by lehug** [comment]
just a try

File: core/app/models/spree/product.rb [hunk 1]
```diff
@@ -33,17 +33,17 @@ class Product < Spree::Base
     has_many :classifications, dependent: :delete_all, inverse_of: :product
     has_many :taxons, through: :classifications, before_remove: :remove_taxon
 
-    has_many :product_promotion_rules, class_name: 'Spree::ProductPromotionRule'
-    has_many :promotion_rules, through: :product_promotion_rules, class_name: 'Spree::PromotionRule'
+    has_many :product_promotion_rules, dependent: :destroy
```
Comment (joneslee85) [inline core/app/models/spree/product.rb]: I think this PR should not introduce any unrelated changes, such as the removal of `class_name` in this case. Doing so, it would help reviewers job easier

**Comment by joneslee85** [comment]
@tommiller1 can you please provide more context to this PR? Plus, can you please add a test too?
